checker/src/main/java/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -163,11 +165,29 @@ private void addSignednessGlbAnnotation(Tree tree, AnnotatedTypeMirror type) {
     }
 
     /**
-     * If the tree is a local variable and the type is byte, short, int, or long, then add the
-     * UnknownSignedness annotation so that dataflow can refine it.
+     * If the tree is in {@link TypeUseLocation#LOCAL_VARIABLE} and the type is byte, short, int,
+     * long, Byte, Short, Integer, or Long, then add {@code @UnknownSignedness} so that dataflow can
+     * refine it.
+     *
+     * @param tree some tree
+     * @param type the tree's type; this method adds {@code @UnknownSignedness} to it
      */
     private void addUnknownSignednessToSomeLocals(Tree tree, AnnotatedTypeMirror type) {
         switch (type.getKind()) {
+            case DECLARED:
+                String qualifiedName =
+                        TypesUtils.getQualifiedName((DeclaredType) type.getUnderlyingType())
+                                .toString();
+                if (qualifiedName.equals("java.lang.Byte")
```

Comment:
We have `isBoxedPrimitive` https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java#L102
and `isIntegral` https://github.com/typetools/checker-framework/blob/master/javacutil/src/main/java/org/checkerframework/javacutil/TypesUtils.java#L238.
The logic here is `isBoxedIntegral`, which we could add to TypesUtils.

Reply:
It's not isBoxedIntegral because it excludes `char`.